src/migrations/tasks/20230116114550_twu_opportunities.ts

```diff
@@ -0,0 +1,154 @@
+  DevOpsSpecialist = "DEVOPS_SPECIALIST"
+}
+
+// eslint-disable-next-line no-empty
+export async function up(connection: Knex): Promise<void> {
+  await connection.schema.createTable("twuOpportunities", (table) => {
+    table.uuid("id").primary().unique().notNullable();
+    table.timestamp("createdAt").notNullable();
+    table.uuid("createdBy").references("id").inTable("users").notNullable();
+  });
+  logger.info("Created twuOpportunities table.");
+
+  await connection.schema.createTable("twuOpportunityVersions", (table) => {
+    table.uuid("id").primary().unique().notNullable();
+    table.timestamp("createdAt").notNullable();
+    table.uuid("createdBy").references("id").inTable("users").notNullable();
+    table
+      .uuid("opportunity")
+      .references("id")
+      .inTable("twuOpportunities")
+      .notNullable()
+      .onDelete("CASCADE");
+    table.text("title").notNullable();
+    table.text("teaser").notNullable();
+    table.boolean("remoteOk").defaultTo(false).notNullable();
+    table.text("remoteDesc").notNullable();
+    table.text("location").notNullable();
+    table.specificType("mandatorySkills", "text ARRAY").notNullable();
+    table.specificType("optionalSkills", "text ARRAY").notNullable();
+    table.text("description").notNullable();
+    table.timestamp("proposalDeadline").notNullable();
+    table.timestamp("assignmentDate").notNullable();
+    table.integer("questionsWeight").notNullable();
+    table.integer("challengeWeight").notNullable();
+    table.integer("priceWeight").notNullable();
+    table.enu("serviceArea", Object.values(TWUServiceArea)).notNullable();
+  });
+  logger.info("Created twuOpportunityVersions table.");
+
+  await connection.schema.createTable("twuOpportunityStatuses", (table) => {
+    table.uuid("id").primary().unique().notNullable();
+    table.timestamp("createdAt").notNullable();
+    table.uuid("createdBy").references("id").inTable("users");
+    table
+      .uuid("opportunity")
+      .references("id")
+      .inTable("twuOpportunities")
+      .notNullable()
+      .onDelete("CASCADE");
+    table.enu("status", Object.values(TWUOpportunityStatus)).nullable();
+    table.enu("event", Object.values(TWUOpportunityEvent)).nullable();
+  });
+
+  await connection.schema.raw(`
+    ALTER TABLE "twuOpportunityStatuses"
+    ADD CONSTRAINT "eitherEventOrStatus" check(
+      ("event" IS NOT NULL AND "status" IS NULL)
+      OR
+      ("event" IS NULL AND "status" IS NOT NULL)
+    )
+  `);
+
+  logger.info("Created twuOpportunityStatuses table.");
+
+  await connection.schema.createTable("twuOpportunityAttachments", (table) => {
+    table.primary(
+      ["opportunityVersion", "file"],
+      "twuOpportunityAttachments_pkey"
+    );
+    table
+      .uuid("opportunityVersion")
+      .references("id")
+      .inTable("twuOpportunityVersions")
+      .notNullable()
+      .onDelete("CASCADE");
+    table.uuid("file").references("id").inTable("files").notNullable();
```

Comment:
Copied this over from the equivalent SWU table, but it got me thinking, shouldn't we cascade this delete for files?

Reply:
If I recall correctly, we don't want to delete from the file table because a hashed file in the database may be used elsewhere (i.e. same file uploaded for multiple opportunities). See `src/back-end/lib/db/file.ts`